lib/components/helpers/array-item-control.js

```diff
@@ -44,12 +44,16 @@ export default React.createClass({
       readOnly: isLastItem && !config.isRemovalOfLastArrayItemAllowed(field)
     });
 
-    const moveItemForward = this.props.index < (this.props.numItems - 1) ?
-      config.createElement('move-item-forward', { field: field, onClick: this.onMoveForward }) :
+    const moveItemForward = field.fieldIndex !== field.parent.value.length - 1 ?
```

Comment:
Thanks for taking a look @jdeal ! Most likely will just switch these back. I remember hunting for a bug last week-- this probably got changed as a result of that (though the bug was elsewhere) and then left in when it worked 😬  but I switched it back and didn't see any problems. Sorry for the confusion! Will push up the change.